src/Raven.Server/Documents/PeriodicBackup/Restore/RestoreBackupTaskBase.cs

```diff
@@ -163,7 +169,48 @@ public static async Task<Stream> CopyRemoteStreamLocally(Stream stream, PathSett
 
                 throw;
             }
+        }
+
+        private static void AssertFreeSpace(Stream stream, string filePath)
+        {
+            long streamLength;
+
+            try
+            {
+                streamLength = stream.Length;
+            }
+            catch (NotSupportedException)
+            {
+                // nothing we can do
+                return;
+            }
+
+            var driveName = DiskUtils.GetDriveInfo(filePath, DriveInfo.GetDrives(), out string _).DriveName;
```

Comment:
This looks like Windows specific code.
We aren't using this on Linux, see:

https://github.com/ravendb/ravendb/blob/52609bb942a26d79b67c6b800ffc63d78144b03e/src/Voron/StorageEnvironmentOptions.cs#L448

Why do you even need this, `GetDiskSpaceInfo` will get a _path_, doesn't need a drive.

Reply:
Because the path doesn't exist yet.
I added this line: var drivesInfo = PlatformDetails.RunningOnPosix ? DriveInfo.GetDrives() : null; , not sure if that's what you meant

Reply:
yes, I can get the space info of the path but its DriveName will be null so it won't have it for the exeption message

Reply:
done.